FileStorage/GettingStarted.cs

```diff
@@ -152,27 +161,27 @@ public async Task RunFileStorageOperationsAsync()
                 // List all files/directories under the root directory.
                 Console.WriteLine("Getting list of all files/directories under the root directory of the share.");
 
-                IEnumerable<IListFileItem> fileList = cloudFileShare.GetRootDirectoryReference().ListFilesAndDirectories();
+                var fileList = rootDirectory.GetFilesAndDirectoriesAsync();
 
                 // Print all files/directories listed above.
-                foreach (IListFileItem listItem in fileList)
+                await foreach (ShareFileItem listItem in fileList)
                 {
-                    // listItem type will be CloudFile or CloudFileDirectory.
-                    Console.WriteLine("    - {0} (type: {1})", listItem.Uri, listItem.GetType());
+                    // listItem type will be ShareClient or ShareDirectoryClient.
+                    Console.WriteLine("    - {0} (type: {1})", listItem.Name, listItem.GetType());
                 }
 
                 Console.WriteLine("Getting list of all files/directories in the file directory on the share.");
 
                 // Now get the list of all files/directories in your directory.
                 // Ordinarily, you'd write something recursive to do this for all directories and subdirectories.
 
-                fileList = fileDirectory.ListFilesAndDirectories();
+                fileList = fileDirectory.GetFilesAndDirectoriesAsync();
 
                 // Print all files/directories in the folder.
-                foreach (IListFileItem listItem in fileList)
+                await foreach (ShareFileItem listItem in fileList)
                 {
-                    // listItem type will be CloudFile or CloudFileDirectory.
-                    Console.WriteLine("    - {0} (type: {1})", listItem.Uri, listItem.GetType());
+                    // listItem type will be ShareClient or ShareDirectoryClient.
+                    Console.WriteLine("    - {0} (type: {1})", listItem.Name, listItem.GetType());
```

Comment:
I think we should use IsDirectory here instead

```suggestion
                    // listItem type will be a file or directory
                    Console.WriteLine("    - {0} (IsDirectory: {1})", listItem.Name, listItem.IsDirectory);
```